tests/foreman/api/test_provisioning.py

```diff
@@ -276,3 +277,145 @@ def test_rhel_ipxe_provisioning(
+
+    :expectedresults:
+        1. Provisioning with host param fips_enabled is successful
+        2. Host installs right version of RHEL
+        3. Satellite is able to run REX job on the fips_enabled host
+        4. Host is registered to Satellite and subscription status is 'Success'
+
+    :parametrized: yes
+
+    :BZ: 2240076
+    """
+    sat = module_provisioning_sat.sat
+    host_mac_addr = provisioning_host._broker_args['provisioning_nic_mac_addr']
+    # Verify password hashing algorithm SHA256 is set in OS used for provisioning
+    assert module_provisioning_rhel_content.os.password_hash == 'SHA256'
+
+    host = sat.api.Host(
+        hostgroup=provisioning_hostgroup,
+        organization=module_sca_manifest_org,
+        location=module_location,
+        name=gen_string('alpha').lower(),
+        mac=host_mac_addr,
+        operatingsystem=module_provisioning_rhel_content.os,
+        subnet=module_provisioning_sat.subnet,
+        host_parameters_attributes=[
+            {
+                'name': 'remote_execution_connect_by_ip',
+                'value': 'true',
+                'parameter_type': 'boolean',
+            },
+            {'name': 'fips_enabled', 'value': 'true', 'parameter_type': 'boolean'},
+        ],
+        build=True,  # put the host in build mode
+    ).create(create_missing=False)
+    # Clean up the host to free IP leases on Satellite.
+    # broker should do that as a part of the teardown, putting here just to make sure.
+    request.addfinalizer(host.delete)
+    # Start the VM, do not ensure that we can connect to SSHD
+    provisioning_host.power_control(ensure=False)
+
+    # TODO: Implement Satellite log capturing logic to verify that
+    # all the events are captured in the logs.
+
+    # Host should do call back to the Satellite reporting
+    # the result of the installation. Wait until Satellite reports that the host is installed.
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
+
+    # Change the hostname of the host as we know it already.
+    # In the current infra environment we do not support
+    # addressing hosts using FQDNs, falling back to IP.
+    provisioning_host.hostname = host.ip
+    # Host is not blank anymore
+    provisioning_host.blank = False
+
+    # Wait for the host to be rebooted and SSH daemon to be started.
+    provisioning_host.wait_for_connection()
+
+    # Perform version check and check if root password is properly updated
+    host_os = host.operatingsystem.read()
+    expected_rhel_version = f'{host_os.major}.{host_os.minor}'
+
+    if int(host_os.major) >= 9:
+        assert (
+            provisioning_host.execute(
+                'echo -e "\nPermitRootLogin yes" >> /etc/ssh/sshd_config; systemctl restart sshd'
+            ).status
+            == 0
+        )
```

Comment:
Can we use 2 different `assert` here. I think assertion will be applicable for second command, what if first command failed to execute..!

Reply:
`/etc/ssh/sshd_config` file must present with expected permissions to user

Reply:
we'll eventually verify this behaviour below when ssh'ing with passwd, so it doesn't make sense to include more assertions